sirepo/job_driver/docker.py

```diff
@@ -104,6 +124,59 @@ def init_class(cls):
     def cpu_slot_peers(self):
         return self.host.instances[self.kind]
 
+    async def _acquire_op_slot(self, op):
+        if self._image_change.in_progress():
+            if op.op_slot:
+                # Prevents deadlock. Another op, x, has set
+                # self._image_change.in_progress. It will clear
+                # self._image_change.in_progress once it gets all resources
+                # This op has a resource op x wants but would otherwise be
+                # blocked by self._image_change.in_progress and not be able to
+                # give it up.
+                self.op_q[op.opName].put_nowait(op.op_slot)
+                op.op_slot = None
+            pkdlog('op={} waiting on _image_change.complete()', op)
+            await self._image_change.complete()
+            raise sirepo.job_supervisor.Awaited()
+        i = self._get_container_image(op.containerImage)
+        if i != self._image:
+            await self._change_image(op, i)
+        await super()._acquire_op_slot(op)
+
+    def _agent_starting_done(self):
+        self._image_change.set_complete()
+        super()._agent_starting_done()
+
+    async def _change_image(self, op, image):
```

Comment:
@robnagler this is the bulk of the code for changing an image. self._image_change was added to control the whole flow of events. I think it could be simpler to just start a new agent with the different image and make sure that no user can have multiple agents allocated a cpu slot.